pkg/cache/internal/cache_reader.go

```diff
@@ -73,14 +73,20 @@ func (c *CacheReader) Get(_ context.Context, key client.ObjectKey, out runtime.O
 	// TODO(directxman12): revisit the decision to always deepcopy
 	obj = obj.(runtime.Object).DeepCopyObject()
 
-	// Copy the value of the item in the cache to the returned value
-	// TODO(directxman12): this is a terrible hack, pls fix (we should have deepcopyinto)
-	outVal := reflect.ValueOf(out)
-	objVal := reflect.ValueOf(obj)
-	if !objVal.Type().AssignableTo(outVal.Type()) {
-		return fmt.Errorf("cache had type %s, but %s was asked for", objVal.Type(), outVal.Type())
+	oType := reflect.TypeOf(obj)
```

Comment:
We could probably do this without reflection if the deepcopy code generated this function.  That should be relatively straightforward to add.  Would you add a TODO for that?

Reply:
Is there not an interface for `DeepCopyInto`?

Reply:
Hey, guys so I don't think that we can use an interface based on how generation works today. 
https://github.com/kubernetes/api/blob/master/core/v1/zz_generated.deepcopy.go#L30

If you notice that the `DeepCopyInto` does not use a `runtime.Object` as its input, it instead uses the type. I believe this means that we can not make a generic interface for `DeepCopyInto`. 

Am I missing something?

Reply:
Sorry, I was suggesting as a follow up we make a function that takes a runtime.Object and checks its type.  It should be trivial to wrap the DeepCopyInto function in another function that takes runtime.Object and does a type check.

Again, that would be a follow up and something we could put in the 1.14 version